src/page-modules/assistant/details/trip-section/index.tsx

```diff
@@ -119,30 +131,33 @@ export default function TripSection({
           </TripRow>
         )}
       </div>
+      {interchangeDetails && leg.interchangeTo?.guaranteed && leg.line && (
+        <div className={style.rowContainer}>
+          <DecorationLine color={unknownTransportationColor.backgroundColor} />
+          <TripRow rowLabel={<MonoIcon icon="actions/Interchange" />}>
+            <MessageBox
+              noStatusIcon
+              type="info"
+              message={
+                leg.line.publicCode
+                  ? t(
+                      PageText.Assistant.details.tripSection.interchange(
+                        leg.line.publicCode,
+                        interchangeDetails.publicCode,
+                        interchangeDetails.fromPlace,
+                      ),
+                    )
+                  : t(
+                      PageText.Assistant.details.tripSection.interchangeWithUnknownFromPublicCode(
+                        interchangeDetails.publicCode,
+                        interchangeDetails.fromPlace,
+                      ),
+                    )
+              }
+            />
+          </TripRow>
+        </div>
+      )}
```

Comment:
Maybe this should be its own component to encapsulate some details and to colocate typings?

Reply:
Also the utils for this specific task

Reply:
I agree 👍 I've updated to reflect this now